File: tasks/2025/round3_pp/dance-pp.py
#!/usr/bin/env python3
# @check-accepted: examples NDsmall Nsmall Dsmall no-limits

N = int(input())
D = int(input())
A = list(map(int, input().split()))
B = list(map(int, input().split()))

ok = True
for i in range(N):
    if B[i]-D <= A[i] < B[i]:
        continue
    else:
        ok = False
        break

if ok:
    print('YES')
else:
    print('NO')

File: tasks/2025/round3_pp/dance-pp.cpp
// @check-accepted: examples NDsmall Nsmall Dsmall no-limits
#include <iostream>
#include <vector>
using namespace std;

int main() {

  int N, D;
  cin >> N >> D;
   
  vector<int> A(N);
  for (int i = 0; i < N; i++) {
    cin >> A[i];
  }

  vector<int> B(N);
  for (int i = 0; i < N; i++) {
    cin >> B[i];
  }

  bool ok = true;
  for (int i = 0; i < N; i++) {
    if (B[i]-D > A[i]) {
      ok = false;
      break;
    }
  }

  if (ok) {
    cout << "YES" << endl;
  } else {
    cout << "NO" << endl;
  }

  return 0;
}
